### PR TITLE
Added FCM Support

### PR DESCRIPTION
**Pull request by AliLozano**
Still missing documentation, but it's ok to start.

### CODE REVIEW

File: push_notifications/fcm.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+			raise FCMError(response)
+	return response
+
+
+def _fcm_handle_canonical_id(canonical_id, current_id):
+	"""
+Handle situation when FCM server response contains canonical ID
+"""
+	if FCMDevice.objects.filter(registration_id=canonical_id, active=True).exists():
+		FCMDevice.objects.filter(registration_id=current_id).update(active=False)
```
Comment (jleclanche) [inline push_notifications/fcm.py]: This doesn't look right.

**Comment by AliLozano** [comment]
I said, its good to start, I didn't find how to send a bulk message with FCM.

**Comment by jleclanche** [comment]
Yeah, I'm leaving the PR open for other people to build upon, but it's not mergeable as-is.

**Comment by jamaalscarlett** [comment]
@AliLozano This looks like a lot of duplicate code.  Couldn't most of it be removed by a new column on the GCM model?  Something like isFCM?

**Comment by AliLozano** [comment]
Yes, It could be a variable in a settings, but i think gcm will be removed in the future.

**Comment by jamaalscarlett** [comment]
@AliLozano  Is this really needed?  This is from the fcm [faq](https://developers.google.com/cloud-messaging/faq) 

How is FCM server implementation different from GCM server implementation?

You won’t need to make any server-side protocol changes for the upgrade. The service protocol has not changed. However, note that all new server enhancements will be documented in FCM server documentation.

Have you tried just changing the url to point to the fcm endpoint and seeing if it works?

**Comment by alexandreferreira** [comment]
Hi I tried an another approach for add FCM in #336 
thanks

**Pull request closed without merging**